File: lib/src/enumerators/notification_action_type.dart
/// Determines what thread will receive the silent action
/// [BringToForeground]: Forces the app to go foreground and run on main thread. Can be interrupt if app gets terminated.
/// [KeepOnTopAction]: Do not forces the app to go foreground (do not closes the status bar in case of buttons), but calls actionStream
/// [DisabledAction]: Do not do anything and don't dismiss the corresponding notification.
/// [SilentAction]: Do not forces the app to go foreground, but runs on main thread, accept visual elements and can be interrupt if main app gets terminated.
// [SilentBackGroundThread]: Do not forces the app to go foreground and runs on background, not accepting any visual elements. The execution is totally
/// apart from app lifecycle and will not be interrupt if the app goes terminated / killed.
enum NotificationActionType {

  /// Forces the app to go foreground and run on main thread. Can be interrupt if app gets terminated.
  BringToForeground,

  /// Do not forces the app to go foreground (do not closes the status bar in case of buttons), but calls actionStream
  KeepOnTopAction,

  /// Do not do anything and don't dismiss the corresponding notification.
  DisabledAction,

  /// Do not forces the app to go foreground, but runs on main thread, accept visual elements and can be interrupt if main app gets terminated.
  SilentAction,

  /// Do not forces the app to go foreground and runs on background, not accepting any visual elements. The execution is totally
  /// apart from app lifecycle and will not be interrupt if the app goes terminated / killed.
  SilentBackgroundAction
}

File: ios/Classes/lib/notifications/enumerators/NotificationActionType.swift
public enum NotificationActionType : String, CaseIterable {
    
    /// Forces the app to go foreground and run on main thread calling actionStream. Can be interrupt if app gets terminated.
    case BringToForeground = "BringToForeground"

    /// Do not forces the app to go foreground and do not closes the status bar, but fires the action event
    case KeepOnTopAction = "KeepOnTopAction"

    /// Do not do fire any action event and only runs on native level.
    case DisabledAction = "DisabledAction"

    /// Do not forces the app to go foreground, but runs silently on main thread, accept visual elements and can be interrupt if main app gets terminated.
    /// It can run while the app is terminated / killed
    case SilentAction = "SilentAction"

    /// Do not forces the app to go foreground and runs silently on background, do not accept any visual elements. The execution is totally
    /// apart from app lifecycle and will not be interrupt if the app goes terminated / killed.
    case SilentBackgroundAction = "SilentBackgroundAction"
    
}
